src/arrays/programs_involving_data_sequences/medium/q7/MainTest.java

```diff
@@ -0,0 +1,61 @@
+package arrays.programs_involving_data_sequences.medium.q7;
+
+import global.BaseTest;
+import global.MethodTest;
+import global.tools.CustomAssertions;
+import global.utils.ArrayUtil;
+import global.variables.Clause;
+import global.variables.clauses.StringLiteral;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+public class MainTest extends BaseTest {
+    //Parsons with distractors
+    public Clause[] testSentence() {
+        return new Clause[]{
+                new StringLiteral("homecominghomecominghomecominghomecominghomecominghomecominghomecominghomecominghomecominghomecoming")
+        };
+    }
+
+    public void runMain() {
+        DupeDupeDupeDupe.main(new String[0]);
+    }
+
+    static Stream<Arguments> inputProvider() {
+        return Stream.of(
```

Comment:
We should also have the test cases of a length 1 array and length 0 array. You can replace some of these other similar test cases with that.